File: python/ListNode.py
from var_dump import var_dump

class ListNode:
    def __init__(self, val=0, next=None):
        self.val = val
        self.next = next

def makeList(li: list[int]) -> list[int]:
    head = ListNode(0)
    h = head
    for i in range(len(li)):
        node = ListNode(li[i])
        h.next = node
        h = h.next
    return head.next


class Solution:
    pass



l1 = [2,4,3]
l2 = [5,6,4]

list1 = makeList(l1)
list2 = makeList(l2)

so = Solution()
r = so.addTwoNumbers(list1, list2)
var_dump(r)

File: php/ListNode.php
<?php

class ListNode {
    public $val = 0;
    public $next = null;
    function __construct($val = 0, $next = null) {
        $this->val = $val;
        $this->next = $next;
    }
}

function makeList($ary) {
    $head = new ListNode($ary[0]);
    $h = $head;
    for ($i=1; $i<count($ary); $i++) {
        $h->next = new ListNode($ary[$i]);
        $h = $h->next;
    }
    return $head;
}

